cpp/src/arrow/compute/kernels/scalar_arithmetic_test.cc

```diff
@@ -2476,6 +2541,19 @@ TYPED_TEST(TestBinaryArithmeticFloating, TrigAtan2) {
                               -M_PI_2, 0, M_PI));
 }
 
+TYPED_TEST(TestUnaryArithmeticFloating, TrigAtanh) {
+  this->SetNansEqual(true);
+  this->AssertUnaryOp(Atanh, "[-Inf, Inf, -2, 2]", "[NaN, NaN, NaN, NaN]");
```

Comment:
add -1 and 1 as expecting NaN as well.

Reply:
That happened a few lines below. But now that they behave differently I've split out the test on lines 2546 and 2553.